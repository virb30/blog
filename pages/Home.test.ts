import Vuex from 'vuex';
import VueMeta from "vue-meta";
import { createLocalVue, mount, RouterLinkStub } from "@vue/test-utils";
import Index from './index.vue';

const localVue = createLocalVue();
localVue.use(VueMeta, { keyName: 'head' });
localVue.use(Vuex);

describe('Home Page', () => {

	let store: any;
	let config: any;

	beforeEach(() => {

		const mockGetter = jest.fn()
		mockGetter.mockImplementationOnce(() => []);

		store = new Vuex.Store({
			modules: {
				Blog: {
					namespaced: true,
					state: {
						posts: [],
						pagination: {},
					},
					getters: {
						last3Posts: mockGetter,
					}
				}
			}
		});

		config = {
			localVue,
			stubs: {
				Profile: true,
				FontAwesomeIcon: true,
				Logo: true,
				NuxtLink: RouterLinkStub,
			},
			store,
		};

	});

	it('should render childs', () => {
		const wrapper = mount(Index, config);

		expect(wrapper.html()).toMatchSnapshot();
	});

	describe('Meta info', () => {
		it('should have a meta title', () => {
			const wrapper = mount(Index, config);

			const expected = 'Home | viniboscoa.dev';

			const meta = wrapper.vm.$meta().refresh();

			expect(meta.metaInfo.title).toBe(expected);
		});
	});
});
