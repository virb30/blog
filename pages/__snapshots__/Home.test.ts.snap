// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render childs 1`] = `
"<main>
  <div class=\\"container w-full bg-cover bg-fixed bg-center flex items-center justify-center flex-col py-16 px-8 text-white\\">
    <profile-stub></profile-stub>
  </div>
  <div class=\\"mx-auto max-w-6xl p-4\\">
    <h2 class=\\"my-4 font-bold text-xl\\">
      Ãšltimos artigos
    </h2>
    <div class=\\"grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3 flex-col sm:flex-row lg:flex-row w-full flex-wrap\\"></div>
    <div class=\\"text-right font-light my-4\\"><a>
        Ver todos os artigos
        <fontawesomeicon-stub icon=\\"fas,chevron-right\\"></fontawesomeicon-stub></a></div>
  </div>
</main>"
`;
